backend-db/Network/PushNotification/Worker.hs

```diff
@@ -79,12 +80,11 @@ apnsWorker cfg delay db = askLoggerIO >>= \logger -> return . killThread <=<
                 if LBS.length (_applePushMessage_payload m) > maxPayloadLength
                   then deleteBy (fromId k) >> clear
                   else do
-                    liftIO $ sendApplePushMessage conn m
+                    unsafeMkSerializable $ liftIO $ sendApplePushMessage conn m
```

Comment:
Can you give some details on what this line does (and why it needs to use "unsafe")?

Reply:
And how it behaves if the transaction is restarted.

Reply:
`unsafeMkSerializable` is just the monad's newtype constructor, but with a scarier name so we don't use it by accident. Since we thought we were already using serializable isolation in Rhyolite I assumed we had thought through the cases already, but that's probably too much to assume.

I suppose I need to know if `sendApplePushMessage` is idempotent. @ali-abrar do you know if this is idempotent? If we're not sure I'll assume it's not and refactor this code to pull it out of the transaction.

Reply:
Actually idempotence isn't enough. The retry could decide not to send it. I'll refactor this.

Reply:
I've reworked this. We need to test it.